R/QueueDataWarehouse.R

```diff
@@ -1,164 +1,206 @@
+#' a blank string or NA if you do not want any date granularity.
+#' @param segment.id Id of Adobe Analytics segment to retrieve the report for
+#' @param data.current TRUE or FALSE - whether to include current data for reports that include today's date
+#' @param expedite Set to TRUE to expedite the processing of this report
+#' @param interval.seconds How long to wait between attempts
+#' @param max.attempts Number of API attempts before stopping
+#' @param validate whether to submit report definition for validation before requesting the data.
+#' @param enqueueOnly only enqueue the report, don't get the data. returns report id, which you can later use to get the data
+#' @param ftp FTP client parameters, only used if enqueueOnly=TRUE. Double check ftp parameters before requesting
+#' a long report.
+#'
+#' @importFrom jsonlite toJSON unbox
+#' @importFrom utils read.csv
+#'
+#' @return Data frame or report id, if enqueueOnly is TRUE
+#'
+#' @examples
+#' \dontrun{
+#' report.data <- QueueDataWarehouse("your_report_suite",
+#'                             "2014-01-01",
+#'                             "2014-01-07",
+#'                             c("visits", "pageviews","event10"),
+#'                             c("page","geoCountry","geoCity"),
+#'                             enqueueOnly=TRUE,
+#'                             ftp = list(host = "myftpserver.com",
+#'                                        port = "21",
+#'                                        directory = "/fromDW/",
+#'                                        username = "memyselfandirene",
+#'                                        password = "valkilmer",
+#'                                        filename = "myreport.csv")
+#'                             )
+#'}
+#' @export
+#' 
+QueueDataWarehouse <- function(reportsuite.id, date.from, date.to, metrics, elements,
+                                   classification = c(), date.granularity='day', segment.id='',
+                                   data.current=TRUE, expedite=FALSE, interval.seconds=5, 
+                                   max.attempts=120, validate=TRUE, enqueueOnly=TRUE, ftp='') {
+  
+  if(enqueueOnly == TRUE && ftp == '') {
+    stop("FTP credentials need to be specified when enqueueOnly = TRUE")
+  }
+  
+  #RZ: move this out of public function interface
+  format <- 'csv'
+  
+  # build JSON description
+  # we have to use unbox to force jsonlite not put strings into single-element arrays
+  report.description <- c()
+  report.description$reportDescription <- c(data.frame(matrix(ncol=0, nrow=1)))
+  report.description$reportDescription$dateFrom <- unbox(date.from)
+  report.description$reportDescription$dateTo <- unbox(date.to)
+  report.description$reportDescription$reportSuiteID <- unbox(reportsuite.id)
+  
+  # no date granularity handling; we need to remove this arg from description so
+  #  set to NULL and Filter out later
+  if(date.granularity == "" || is.na(date.granularity)) {

```

Comment:
Is this just a syntactic sugar improvement? Any reason to not just highlight that `NULL` is also appropriate in the documentation?

Reply:
Mainly an end-user consideration. Since there is a default, I figured an empty string or NA would seem most natural to denote "none". Now that I think about it, we could just use "none" and cast that to NULL in the body? Or also OK with simply expecting NULL only to have no date granularity, and let toJSON filter it out.